lib/archiver.js

```diff
@@ -0,0 +1,168 @@
+var path = require('path')
+var promisify = require('es6-promisify')
+var hyperdrive = require('hyperdrive')
+var datEncoding = require('dat-encoding')
+var discoverySwarm = require('discovery-swarm')
+var swarmDefaults = require('datland-swarm-defaults')
+var mkdirp = require('mkdirp')
+var debug = require('debug')('archiver')
+
+mkdirp = promisify(mkdirp)
+
+// exported api
+// =
+
+module.exports = class Archiver {
+  constructor (config) {
+    this.config = config
+    this.archives = {}
+    this.loadPromises = {}
+  }
+
+  // methods
+  // =
+
+  getArchive (key) {
+    return this.archives[key]
+  }
+
+  isLoadingArchive (key) {
+    return key in this.loadPromises
+  }
+
+  // load archive (wrapper) manages load promises
+  async loadArchive (key) {
+    key = datEncoding.toStr(key)
+
+    // fallback to archive if it exists
+    if (key in this.archives) {
+      return this.archives[key]
```

Comment:
Yeah, we're eventually going to need to put them into some kind of rotation, but I'm not sure what the logic will be. We need to keep them active to receive updates.